Add Video Calling/iOSVideo/CallKitHelper.swift

```diff
@@ -0,0 +1,553 @@
+                        incomingCall.accept(options: acceptCallOptions ?? AcceptCallOptions(), completionHandler: completionBlock)
+                    } else {
+                        completionBlock(nil, CallKitErrors.noIncomingCallFound)
+                    }
+                }
+            }
+        }
+    }
+
+    func provider(_ provider: CXProvider, perform action: CXEndCallAction) {
+        Task {
+            guard let activeCall = await self.callKitHelper.getActiveCall(callId: action.callUUID.uuidString) else {
+                action.fail()
+                return
+            }
+
+            let activeCallInfo = await self.callKitHelper.getActiveCallInfo(transactionId: action.uuid.uuidString)
+            activeCall.hangUp(options: nil) { error in
+                // Its ok if hangup fails because we maybe hanging up already hanged up call
+                action.fulfill()
+                activeCallInfo?.completionHandler(error)
+                Task {
+                    await self.callKitHelper.removeActiveCall(callId: activeCall.id)
+                    await self.callKitHelper.removeActiveCallInfo(transactionId: action.uuid.uuidString)
+                }
+            }
+        }
+    }
+
+    func provider(_ provider: CXProvider, didActivate audioSession: AVAudioSession) {
+        Task {
+            guard let activeCall = await self.callKitHelper.getActiveCall() else {
+                print("No active calls found !!")
+                return
+            }
+
+            try await startAudio(call: activeCall)
+        }
+    }
+
+    func provider(_ provider: CXProvider, perform action: CXStartCallAction) {
+        Task {
+            // This will be raised by CallKit always after raising a transaction
+            // Which means an API call will have to happen to reach here
+            guard let outInCallInfo = await callKitHelper.getOutInCallInfo(transactionId: action.uuid) else {
+                return
+            }
+            
+            let completionBlock : ((Call?, Error?) -> Void) = { (call, error) in
+                
+                if error == nil {
+                    action.fulfill()
+                    Task {
+                        await self.callKitHelper.addActiveCall(callId: action.callUUID.uuidString,
+                                                               call: call!)
+                    }
+                } else {
+                    action.fail()
+                }
+                outInCallInfo.completionHandler(call, error)
+                Task {
+                    await self.callKitHelper.removeOutInCallInfo(transactionId: action.uuid)
+                }
+            }
+
+            guard let callAgent = self.callAgent else {
+                completionBlock(nil, CallKitErrors.noCallAgent)
+                return
+            }
+
+            let error = configureAudioSession()
+
+            if error == nil {
```

Comment:
This could be `if let error = configureAudioSession() { <handle error> } else { <do the other long thing> }`
It's generally clearer to read to do the smaller of the two conditions first, especially in the case of a single call and return. Bonus points for using `guard`

Reply:
I missed that , this can be changed.